addons/block_code/ui/picker/categories/category_factory.gd

```diff
@@ -191,20 +203,27 @@ static func get_general_blocks() -> Array[Block]:
 	b = BLOCKS["entry_block"].instantiate()
 	# HACK: make signals work with new entry nodes. NIL instead of STRING type allows
 	# plain text input for function name. Should revamp signals later
-	b.block_format = "On signal {signal: NIL}"
-	b.statement = "func signal_{signal}():"
+	b.block_format = "Define method {method_name: NIL}"
+	b.statement = "func {method_name}():"
 	b.category = "Signal"
 	block_list.append(b)
 
 	b = BLOCKS["statement_block"].instantiate()
-	b.block_format = "Send signal {signal: STRING} to group {group: STRING}"
-	b.statement = 'if get_tree().root.has_node("SignalManager"):\n' + '\tget_tree().root.get_node_or_null("SignalManager").broadcast_signal({group}, {signal})'
+	b.block_format = "Call method {method_name: STRING} in group {group: STRING}"
+	b.statement = "get_tree().call_group({group}, {method_name})"
 	b.category = "Signal"
 	block_list.append(b)
 
 	b = BLOCKS["statement_block"].instantiate()
-	b.block_format = "Send signal {signal: STRING} to node {node: NODE_PATH}"
-	b.statement = 'if get_tree().root.has_node("SignalManager"):\n' + '\tget_tree().root.get_node_or_null("SignalManager").send_signal_to_node({node}, {signal})'
+	b.block_format = "Call method {method_name: STRING} in node {node_path: NODE_PATH}"
+	b.statement = (
```

Comment:
Yeah. After all the type discussions I realize that we should have 2 types of Value blocks:
- Variant block
- Node block
The Value node blocks that can have a hint like "Node2D" for parameters. Basically the PROPERTY_HINT_NODE_TYPE in global scope. But yes let's think about this as a group after the MVP.

Reply:
> This generates code that looks like this:
> 
> ```
> func _on_body_entered(_body: Node2D):
> 	var body: NodePath = _body.get_path()
> 
> 	var node = get_node(body)
> 	if node:
> 		node.call('test')
> ```
> 
> We should probably have both the signal and the "call method in node" block work with `TYPE_OBJECT` instead of `TYPE_NODE_PATH`. We can do this in another PR.

By the way is this something that have regressed? I think it was like this before the multi-line change.

Reply:
Aha, yes, that would be good. Maybe we don't have to make a new block since a node block is really a value too? We could just add a new property to value block or extend it.

We'll discuss!